onnxruntime/core/optimizer/initializer.cc

```diff
@@ -27,10 +27,14 @@ Initializer::Initializer(ONNX_NAMESPACE::TensorProto_DataType data_type,
 
 Initializer::Initializer(const ONNX_NAMESPACE::TensorProto& tensor_proto, const Path& model_path) {
   ORT_ENFORCE(utils::HasDataType(tensor_proto), "Initializer must have a datatype");
+#if !defined(__wasm__)
+  // using full filepath is required by utils::TensorProtoToTensor(). One exception is WebAssembly platform, where
+  // external data is not loaded from real file system.
   if (utils::HasExternalData(tensor_proto)) {
     ORT_ENFORCE(!model_path.IsEmpty(),
                 "model_path must not be empty. Ensure that a path is provided when the model is created or loaded.");
   }
+#endif
```

Comment:
@HectorSVC, @jywu-msft 

Do you think this change is reasonable, or I should wait for #19089 to merge and set a dummy file path for web usage?